src/test/java/org/junitpioneer/jupiter/resource/TemporaryDirectoryTests.java

```diff
@@ -0,0 +1,496 @@
+			assertReadableWriteableTemporaryDirectory(tempDir);
+			assertCanAddAndReadTextFile(tempDir);
+
+			recordedPath = tempDir;
+		}
+
+	}
+
+	// ---
+
+	@DisplayName("when two test classes in the same file have a test method with a "
+			+ "@Shared(factory = TemporaryDirectory.class, name = \"some-name\", scope = SOURCE_FILE)-annotated "
+			+ "parameter")
+	@Nested
+	class WhenTwoTestClassesInSameFileHaveTestMethodWithParameterWithSameNamedAndSourceFileScopedSharedTempDirTests {
+
+		@DisplayName("then the parameters are populated with a shared readable and writeable "
+				+ "temporary directory that is torn down afterwards")
+		@Test
+		void thenParametersArePopulatedWithSharedReadableAndWriteableTempDirsThatIsTornDownAfterwards() {
+			ExecutionResults executionResults = PioneerTestKit
+					.executeTestClass(SingleTestMethodWithSourceFileScopedTempDirParameter.TestCases.class);
+			assertThat(executionResults).hasNumberOfSucceededTests(2);
+			assertThat(SingleTestMethodWithSourceFileScopedTempDirParameter.recordedPathForImplicit)
+					.isEqualTo(SingleTestMethodWithSourceFileScopedTempDirParameter.recordedPathForExplicit);
+			assertThat(SingleTestMethodWithSourceFileScopedTempDirParameter.recordedPathForImplicit).doesNotExist();
+			assertThat(SingleTestMethodWithSourceFileScopedTempDirParameter.recordedPathForExplicit).doesNotExist();
+		}
+
+	}
+
+	// ---
+
+	@DisplayName("when two test classes in different files have a test method with a "
+			+ "@Shared(factory = TemporaryDirectory.class, name = \"some-name\", scope = SOURCE_FILE)-annotated "
+			+ "parameter")
+	@Nested
+	class WhenTwoTestClassesInDiffFilesHaveTestMethodWithParameterWithSameNamedAndSourceFileScopedSharedTempDirTests {
+
+		@DisplayName("then the parameters are populated with unique readable and writeable "
+				+ "temporary directories that that torn down afterwards")
+		@Test
+		void thenParametersArePopulatedWithUniqueReadableAndWriteableTempDirsThatAreTornDownAfterwards() {
+			ExecutionResults executionResults = PioneerTestKit
+					.executeTestClasses( //
+						asList( //
+							FirstSingleTopLevelTestMethodWithSourceFileScopedTempDirParameter.OnlyTestCases.class,
+							SecondSingleTopLevelTestMethodWithSourceFileScopedTempDirParameter.OuterTestCases.class));
+
+			assertThat(executionResults).hasNumberOfSucceededTests(3);
+
+			assertThat(FirstSingleTopLevelTestMethodWithSourceFileScopedTempDirParameter.recordedPath)
+					.isNotEqualTo(SecondSingleTopLevelTestMethodWithSourceFileScopedTempDirParameter.recordedOuterPath);
+			assertThat(FirstSingleTopLevelTestMethodWithSourceFileScopedTempDirParameter.recordedPath)
+					.isNotEqualTo(SecondSingleTopLevelTestMethodWithSourceFileScopedTempDirParameter.recordedInnerPath);
+			assertThat(SecondSingleTopLevelTestMethodWithSourceFileScopedTempDirParameter.recordedOuterPath)
+					.isEqualTo(SecondSingleTopLevelTestMethodWithSourceFileScopedTempDirParameter.recordedInnerPath);
```

Comment:
Isn't this a repetition of the previous tests, where you already verified that source-file scope works? This one is slightly different (with inner and outer class), but if you feel it's necessary to test that, I recommend to make that a separate test.

Reply:
Oh, good catch. Yeah, I can't remember why I went so far as to test with `@Nested` test classes here, so I've eliminated the duplication in commit https://github.com/junit-pioneer/junit-pioneer/pull/491/commits/fedd6d49a7720b0f366f232f613d3c94428d3b28.